### PR TITLE
Complete type definitions from JS migration

### LINKED ISSUE #1: Complete type definitions
The [JavaScript client PR](https://github.com/weaviate/weaviate-javascript-client/pull/121) which migrated this client from untyped javascript included some types, but not all of them. Now there are a remainder of any types which should be assigned their proper types, where possible.

A quick way to determine where these changes need to be made can be done by grepping the repo for `: any`.

### PR DESCRIPTION
**Pull request by parkerduckworth**
Closes #1, #15

### CODE REVIEW

File: src/data/creator.ts [hunk 1]
```diff
@@ -2,21 +2,22 @@ import { isValidStringProperty } from '../validation/string';
...
 import { ObjectsPath } from './path';
 import { CommandBase } from '../validation/commandBase';
+import { Properties, WeaviateObject } from '../openapi/types';
 
 export default class Creator extends CommandBase {
   private className?: string;
   private consistencyLevel?: string;
   private id?: string;
   private objectsPath: ObjectsPath;
   private properties?: any;
```
Comment (antas-marcin) [inline src/data/creator.ts:12]: Here do we have some sort of a type for `Properties`?

File: src/graphql/explorer.ts [hunk 2]
```diff
@@ -32,7 +32,7 @@ export default class Explorer extends CommandBase {
     return this;
   };
 
-  withNearText = (nearTextObj: any) => {
+  withNearText = (nearTextObj: object) => {
```
Comment (antas-marcin) [inline src/graphql/explorer.ts:35]: What is `object`? shouldn't be here `NearText`?

File: src/graphql/getter.ts [hunk 3]
```diff
@@ -68,15 +69,13 @@ export default class Getter extends CommandBase {
     return this;
   };
 
-  withNearText = (nearTextObj: any) => {
+  withNearText = (args: object) => {
```
Comment (antas-marcin) [inline src/graphql/getter.ts:72]: here also, shouldn't be here `NearTexr` instead of `object`?

File: src/graphql/getter.ts [hunk 4]
```diff
@@ -165,17 +158,17 @@ export default class Getter extends CommandBase {
...
   };
 
-  withOffset = (offset: any) => {
+  withOffset = (offset: number) => {
     this.offset = offset;
     return this;
   };
 
-  withSort = (sortObj: any) => {
+  withSort = (sortObj: object) => {
```
Comment (antas-marcin) [inline src/graphql/getter.ts:171]: Do we have a type (or interface) for `Sort`?

File: src/data/deleter.ts [hunk 5]
```diff
@@ -1,13 +1,14 @@
 import Connection from '../connection';
 import { CommandBase } from '../validation/commandBase';
+import { ObjectsPath } from './path';
 
 export default class Deleter extends CommandBase {
-  private className?: string;
+  private className!: string;
   private consistencyLevel?: string;
```
[the remark this answers is not in the document]
Comment (parkerduckworth) [reply src/data/deleter.ts]: Good idea, I replaced all the other constants with enums as well

File: src/data/creator.ts (hunk 1 above)
Comment (parkerduckworth) [reply src/data/creator.ts]: Yes, fixed

File: src/graphql/getter.ts (hunk 3 above)
Comment (parkerduckworth) [reply src/graphql/getter.ts]: Updated with interface for NearTextArgs

File: src/graphql/getter.ts (hunk 4 above)
Comment (parkerduckworth) [reply src/graphql/getter.ts]: Updated with interface for Sort

File: src/graphql/explorer.ts (hunk 2 above)
Comment (parkerduckworth) [reply src/graphql/explorer.ts]: Updated with interface for NearText args

File: src/connection/auth.ts [hunk 6]
```diff
@@ -77,16 +65,14 @@ export class OidcAuthenticator {
...
-      .externalGet(localConfig.href)
-      .then((openidProviderConfig: any) => {
-        const scopes = localConfig.scopes || [];
-        return {
-          clientId: localConfig.clientId,
-          provider: openidProviderConfig,
-          scopes: scopes,
-        };
-      });
+    return this.http.externalGet(localConfig.href).then((openidProviderConfig: any) => {
```
[the remark this answers is not in the document]
Comment (parkerduckworth) [reply src/connection/auth.ts:68]: No, this is just and object which depends on the individual openID provider response

File: src/data/path.ts [hunk 7]
```diff
@@ -13,65 +14,45 @@ export class ObjectsPath {
...
-      ]
-    );
+    return this.build({ id, className, additional: additional, consistencyLevel, nodeName }, [
+      this.addClassNameDeprecatedNotSupportedCheck,
+      this.addId,
+      this.addQueryParams,
+    ]);
   }
   buildGet(
     className: string | undefined,
```
Comment (antas-marcin) [inline src/data/path.ts:41]: could be replaced with `className?: string`

File: src/batch/objectsBatchDeleter.ts [hunk 8]
```diff
@@ -2,12 +2,15 @@ import { isValidStringProperty } from '../validation/string';
...
+import { DeleteOutput } from '.';
 
 export default class ObjectsBatchDeleter extends CommandBase {
   private className?: string;
-  private consistencyLevel?: string;
+  private consistencyLevel?: ConsistencyLevel;
   private dryRun?: boolean;
-  private output?: any;
+  private output?: DeleteOutput;
   private whereFilter?: any;
```
[the remark this answers is not in the document]
Comment (parkerduckworth) [reply src/batch/objectsBatchDeleter.ts:14]: Good catch, added type

File: src/data/path.ts (hunk 7 above)
Comment (parkerduckworth) [reply src/data/path.ts]: Yep, updated

**Pull request merged**